<!doctype html>
<html>
	<head>
		<meta charset="utf-8">
		<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">

		<title>fDOG presentation</title>

		<link rel="stylesheet" href="css/reveal.css">
		<link rel="stylesheet" href="css/theme/beige.css" id=theme>

		<!-- Theme used for syntax highlighting of code -->
		<link rel="stylesheet" href="lib/css/zenburn.css">

		<!-- Printing and PDF exports -->
		<script>
			var link = document.createElement( 'link' );
			link.rel = 'stylesheet';
			link.type = 'text/css';
			link.href = window.location.search.match( /print-pdf/gi ) ? 'css/print/pdf.css' : 'css/print/paper.css';
			document.getElementsByTagName( 'head' )[0].appendChild( link );
		</script>
	</head>
	<body>
		<div class="reveal">
			<div class="slides">

				<!-- SLIDE 0 title -->
				<section data-background-image="images/00_logo.png" data-background-position="10% 90%" data-background-size="10%">
					<h2>fDOG</h2>
					<p style="font-size:40px"><strong><span style="color: #008000;">f</span>eature-aware <span style="color: #008000;">D</span>irected <span style="color: #008000;">O</span>rtholo<span style="color: #008000;">G</span> search</strong></p>
					<hr>
					<h4>22.03.2021</h4>
					<h4>Vinh Tran</h4>
				</section>

				<!-- SLIDE 1 orthology -->
				<section>
					<h5 style="color: white; background-color: #559951"><strong>Orthology</strong></h5>
					<section data-transition="none">
						<p>Homology</p>
						<img src="images/011_homology.png" style='float: top; border:0; box-shadow: none' height=250/>
						<p>
							<ul>
								<li style="font-size:20px">Paralogs <- duplication</li>
								<li style="font-size:20px">Orthologs <- speciation</li>
							</ul>
						</p>
					</section>
					<section data-transition="none">
						<p>Applications</p>
						<p>
							<li style="font-size:20px; text-align:left">Phylogenetic profiles</li>
							<img src="images/012_phyloprofile.png" style='float: top; border:0; box-shadow: none' height=150/>
							<li style="font-size:20px; text-align:left">Gene tree reconstruction, Tree reconciliation</li>
							<a>
								<img src="images/013_tree_reconciliation.png" style='float: top; border:0; box-shadow: none' height=150/>
								<div class="caption" style="font-size:7px">Mallo D. et al. (2016), Syst Biol</div>
							</a>
							<li style="font-size:20px; text-align:left">Functions transfering</li>
							<li style="font-size:20px; text-align:left">...</li>
						</p>
					</section>
					<section data-transition="none">
						<p>RBH-based method</p>
						<p>
							<a>
								<img src="images/014_orthomcl.png" style='float: top; border:0; box-shadow: none' height=250/>
								<div class="caption" style="font-size:7px">Chen et al. (2007), Plos ONE</div>
							</a>
						</p>
					</section>
					<aside class="notes">
						Oh hey, these are some notes. They'll be hidden in your presentation, but you can see them if you open the speaker notes window (hit »S« on your keyboard).
					</aside>
				</section>

				<!-- SLIDE 2 fdog origin -->
				<section>
					<h5 style="color: white; background-color: #559951"><strong>The origin of fDOG</strong></h5>
					<section data-transition="none">
						<p>HaMStR</p>
						<img src="images/021_hamstr_paper.png" style='float: top; border:0; box-shadow: none' height=150/>
						<p>
							<img src="images/021_hamstr.gif" style='float: top; border:0; box-shadow: none' height=120/>
						</p>
					</section>
					<section data-transition="none">
						<p>HaMStR</p>
						<p>
							<img src="images/022_hamstr.gif" style='float: top; border:0; box-shadow: none' height=120/>
						</p>
						<img src="images/022_hamstr_workflow.png" style='float: top; border:0; box-shadow: none' height=350/>
					</section>
					<section data-transition="none">
						<p>
							<div style="width:500px; margin:0 auto;">
								<img src="images/023_carrot.gif" style='float: top; border:0; box-shadow: none' height=120/>
								+
								<img src="images/021_hamstr.gif" style='float: top; border:0; box-shadow: none' height=120/>
								<div style="float:center; width:100 px"><img src="images/023_arrow.png" style='float: center; border:0; box-shadow: none' height=100/></div>
							</div>
						</p>
						<p>
							<img src="images/023_dog.gif" style='float: top; border:0; box-shadow: none' height=180/>
						<br>fDOG</p>
					</section>
					<aside class="notes">
						Oh hey, these are some notes. They'll be hidden in your presentation, but you can see them if you open the speaker notes window (hit »S« on your keyboard).
					</aside>
				</section>

				<!-- SLIDE 3 fdog workflow -->
				<section>
					<h5 style="color: white; background-color: #559951"><strong>How fDOG works</strong></h5>
					<section data-transition="none">
						<img src="images/031_fdog_workflow.jpg" style='float: top; border:0; box-shadow: none' height=550/>
					</section>
					<section data-transition="none">
						<img src="images/032_fdog_workflow.jpg" style='float: top; border:0; box-shadow: none' height=550/>
					</section>
					<section data-transition="none">
						<img src="images/033_fdog_workflow.jpg" style='float: top; border:0; box-shadow: none' height=550/>
					</section>
					<aside class="notes">
						Oh hey, these are some notes. They'll be hidden in your presentation, but you can see them if you open the speaker notes window (hit »S« on your keyboard).
					</aside>
				</section>

				<!-- SLIDE 4 comparison -->
				<section>
					<h5 style="color: white; background-color: #559951"><strong>How good is fDOG</strong></h5>
					<section data-transition="none">
						<p>Benchmark data</p>
						<img src="images/041_qfo.png" style='float: top; border:0' height=250/>
						<p>
							<ul>
								<li style="font-size:20px">78 taxa QfO 2018</li>
								<li style="font-size:20px">Human as reference species</li>
								<li style="font-size:20px">20,996 human proteins as seed sequences</li>
							</ul>
						</p>
					</section>
					<section data-transition="none">
						<p>Benchmark method</p>
						<p><img src="images/042_benchmark_service.png" style='float: top; border:0' height=50/><br>
						<img src="images/043_benchmark_service.png" style='float: top; border:0; box-shadow: none' height=350/><br>
						</p>
					</section>
					<section data-transition="none">
						<img src="images/044_count.png" style='float: top; border:0; box-shadow: none' height=250/>
						<ul>
							<li style="font-size:15px">Compared using Quest for Orthologs - Challenge 6</li>
						</ul>
					</section>
					<section data-transition="none">
						<img src="images/045_ec.png" style='float: left; border:0; box-shadow: none' height=200/>
						<img src="images/046_go.png" style='float: right; border:0; box-shadow: none' height=200/ class="fragment roll-in">
						<img src="images/047_tree.png" style='float: bottom; border:0; box-shadow: none' height=200/ class="fragment roll-in">
					</section>

					<aside class="notes">
						Oh hey, these are some notes. They'll be hidden in your presentation, but you can see them if you open the speaker notes window (hit »S« on your keyboard).
					</aside>
				</section>

				<!-- SLIDE 5 core groups -->
				<section>
					<h5 style="color: white; background-color: #559951"><strong>How meaningful are the core ortholog groups</strong></h5>
					<section data-transition="none">
						<img src="images/051_qfotree.png" style='float: top; border:0; box-shadow: none' height=350/>
						<p>... in reflecting the species relationship</p>
					</section>
					<section data-transition="none">
						<ul>
							<li style="font-size:20px">
								<p>Topology test using <a target="_blank" href="http://www.iqtree.org">IQ-TREE</a></p>
								<img src="images/052_testOutput.png" style='float: top; border:0; box-shadow: none' height=150/>
								<p style="font-size:15px"><i>(Topology rejected (-) when p-AU < 0.05, otherwise accepted (+))</i></p>
							</li>
							<li style="font-size:20px" class="fragment roll-in">
								<p>18,067 core groups with more than 3 sequences</p>
								<img src="images/053_coreSize.png" style='float: top; border:0; box-shadow: none' height=200/>
							</li>
						</ul>
						</p>
					</section>
					<section data-transition="none">
						<table style="font-size:15px">
							<tr>
								<th>Approach</th>
								<th>Accepted</th>
								<th>Rejected</th>
							</tr>
							<tr>
								<td>No correction</td>
								<td>11,707 (64.80%)</td>
								<td>6,360 (35.20%)</td>
							</tr>
							<tr>
								<td>Benjamini + Hochberg (BH) correction</td>
								<td>13,625 (75.41%)</td>
								<td>4,442 (24.59%)</td>
							</tr>
							<tr>
								<td>Bonferroni correction</td>
								<td>16,204 (88.69%)</td>
								<td>2,043 (11.31%)</td>
							</tr>
						</table>
						<img src="images/054_topoStat.png" style='float: top; border:0; box-shadow: none' height=300>
					</section>
					<section data-transition="none">
						<img src="images/055_pvalue_vs_dlogl.png" style='float: top; border:0; box-shadow: none' height=350>
					</section>
					<section data-transition="none">
						<img src="images/056_concord.png" style='float: top; border:0; box-shadow: none' height=300>
					</section>
				</section>

				<!-- SLIDE 6 runtime -->
				<section>
					<h5 style="color: white; background-color: #559951"><strong>How fast is fDOG</strong></h5>
					<section data-transition="none">
						<table style="font-size:15px">
							<tr>
								<th>Data set</th>
								<th>Core compilation</th>
								<th>Ortholog search</th>
								<th>FAS calculation</th>
							</tr>
							<tr>
								<td>20,995 proteins (*)</td>
								<td>19 hrs</td>
								<td>67.5 hrs</td>
								<td>21 hrs</td>
							</tr>
							<tr>
								<td>Q8WZ42 (**)</td>
								<td>54 hrs</td>
								<td>0.2 hrs</td>
								<td>20.5 hrs</td>
							</tr>
						</table>
						<p style="font-size:12px" align = "right"><i>
							(*) <b>fdogs.run</b> using 64 CPUs on compute17 (2x AMD EPIC 7601 2.2Ghz, 32 cores)<br>
							(**) <b>fdog.run</b> using 1 CPU on compute12 (intel i7-6700K 4.0Ghz)
						</i></p>
					</section>
					<section data-transition="none">
						<img src="images/061_hmsruntime.png" style='float: top; border:0; box-shadow: none' height=250>
						<br>
						<p style="font-size:20px">On average, each human protein needs:</p>
						<!-- <br> -->
						<ul style="font-size:15px">
							<li>
								<p>212s for the core compilation</p>
							</li>
							<li>
								<p>10s for the ortholog search</p>
							</li>
						</ul>
						<p style="font-size:10px"><i>Note: number of core taxa and search taxa are the same (78 QfO taxa)</i></p>
					</section>
					<section data-transition="none">
						<img src="images/062_hmsruntime_vsLen.png" style='float: top; border:0; box-shadow: none' width=650/>
						<div class="caption" style="font-size:15px"><i>Pearson p-value < 2.2e-16<br>Correlation coefficient 0.61 for core compilation and 0.24 for ortholog search</i></div>
					</section>
					<section data-transition="none">
						<p style="color: #5e9ca0; font-size:20px">fdog.run<span style="color: #000000;"> VS</span> <span style="color: #ff0000;">fdogs.run</span>
						<br>
							<div style="width:500px; margin:0 auto;">
								<img src="images/063_fdog.gif" style='float: left; border:0; box-shadow: none' height="150"/>
								<img src="images/063_fdogs.gif" style='float: right; border:0; box-shadow: none' height="150"/>
							</div>
						</p>
						<div class="fragment roll-in">
							<img src="images/063_fdog_vs_fdogs.png" style='float: top; border:0; box-shadow: none' width=550/>
							<p style="font-size:10px"><i>Runtime of 6 randomly selected proteins using 4 CPUs</i></p>
						</div>
					</section>
				</section>

				<!-- SLIDE 6 case study -->
				<section>
					<h5 style="color: white; background-color: #559951"><strong>An use-case: Where does the archaea locate</strong></h5>
					<section data-transition="none">
						<p><img src="images/071_hypotheses.png" style='float: top; border:0; box-shadow: none' height=350/></p>
						<div class="caption" style="font-size:7px">Eme et al. (2017), Nat. Reviews Microbiology</div>
						<ol style="list-style-type: lower-alpha;">
							<li style="font-size:15px" class="fragment roll-in">The well-known three-domains tree of life</li>
							<li style="font-size:15px" class="fragment roll-in">Two-domains tree of life</li>
							<li style="font-size:15px" class="fragment roll-in">Eukaryota branches as sister to TACK superphylum</li>
							<li style="font-size:15px" class="fragment roll-in">Eukaryota originated from within (or as sister to) the Asgard archaea</li>
						</ol>
						<aside class="notes">
							A schematic representation of our understanding of the relationships between eukaryotes and archaea over the past 40 years. In a three-domains tree of life (part a), Archaea and Eukarya each represent a monophyletic group and share a unique common ancestor to the exclusion of Bacteria2,3. By contrast, a two-domain tree of life was proposed early on, and the topology of this tree has undergone several changes.When the Archaea was thought to consist of the Euryarchaeota and the Crenarchaeota phyla only (part b), several kinds of molecular evidence supported the close relationship of eukaryotes and Crenarchaeota6,7. In the early 2010s, eukaryotes were found to branch within, or as sister to, the Thaumarchaeota, Aigarchaeota, Crenarchaeota and Korarchaeota (TACK) superphylum25,28–32(part c). Phylogenomic analyses that included members of the Asgard superphylum strongly suggested that eukaryotes originated from within the Asgard archaea or that they represented a sister group to them47,52(part d). Bacteria and Eukarya are indicated in light purple and red, respectively, whereas green and blue represent archaeal lineages. Domain and superphylum rank-level names are bolded or italicized, respectively. Names of proposed phyla are in quotation marks. Note that the Diapherotrites, Parvarchaeota, Aenigmarchaeota, Nanoarchaeota and Nanohaloarchaeota (DPANN) lineage is represented as a monophyletic lineage, although this is a topic of debate
						</aside>
					</section>
					<section data-transition="none">
						<img src="images/072_tree_reconstruction.jpg" style='float: top; border:0; box-shadow: none' height=400/>
					</section>
					<section data-transition="none">
						<img src="images/073_phyloprofile.png" style='float: top; border:0; box-shadow: none' height=180/>
						<ul>
							<li style="font-size:15px" class="fragment roll-in">48 core genes (FAS >= 0.75, present in at least 50% taxa of each superkingdom)</li>
							<li style="font-size:15px" class="fragment roll-in">Super-alignment of 139 sequences, 13,060 aa in length</li>
							<li style="font-size:15px" class="fragment roll-in">Tree reconstruction using IQ-TREE</li>
							<ul style="list-style-type: circle;" class="fragment roll-in">
								<li style="font-size:12px"><strong>-m MFP</strong>: auto identify best fitting model with ModelFinder => LG+R9</li>
								<li style="font-size:12px"><strong>-bb 1000</strong>: ultrafast bootstrap approximation with 1000 replicates</li>
								<li style="font-size:12px"><strong>-bnni</strong>: reduce the risk of overestimating branch supports with UFBoot</li>
							</ul>
						</ul>
						<aside class="notes">
							LG substitution with FreeRate model
						</aside>
					</section>
					<section data-transition="none">
						<p><img src="images/074_asgard_tree.jpg" style='float: top; border:0; box-shadow: none' height=600/></p>
					</section>
					<section data-transition="none">
						<p><img src="images/071_hypotheses.png" style='float: right; border:0; box-shadow: none' height=300/ class="fragment roll-in"></p>
						<p><img src="images/075_asgard_tree_schematic.jpg" style='float: left; border:0; box-shadow: none' height=300/></p>
					</section>
				</section>
				<!-- SLIDE 8 links-->
				<section>
					<h5 style="color: white; background-color: #559951"><strong></strong></h5>
					<section>
						<img src="images/081_thanks.gif" style='float: top; border:0' height="250"/>
						<br>
						<img src="images/082_qrcode.png" style='float: top; border:0' height="100"/>
					</section>
				</section>

			</div>
		</div>

		<script src="lib/js/head.min.js"></script>
		<script src="js/reveal.js"></script>

		<script>
			// More info about config & dependencies:
			// - https://github.com/hakimel/reveal.js#configuration
			// - https://github.com/hakimel/reveal.js#dependencies
			Reveal.initialize({
				controls: true,
				progress: true,
				history: true,
				center: true,
				backgroundTransition: 'zoom', // Not working?!?!?

				dependencies: [
					{ src: 'lib/js/classList.js', condition: function() { return !document.body.classList; } },
		      { src: 'plugin/markdown/marked.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
		      { src: 'plugin/markdown/markdown.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
		      { src: 'plugin/zoom-js/zoom.js', async: true, condition: function() { return !!document.body.classList; } },
		      { src: 'plugin/notes/notes.js', async: true, condition: function() { return !!document.body.classList; } }
				]
			});
		</script>
	</body>
</html>
